lib/shared/sdk/standard/index.js

```diff
@@ -17,48 +17,77 @@
 'use strict'
 
 const Bluebird = require('bluebird')
+const EventEmitter = require('events')
 const drivelist = Bluebird.promisifyAll(require('drivelist'))
 
 /**
- * @summary The name of this adaptor
- * @public
- * @type {String}
- * @constant
+ * @summary StandardAdapter
+ * @class
  */
-exports.name = 'standard'
+class StandardAdapter extends EventEmitter {
+  /**
+   * @summary StandardAdapter constructor
+   * @class
+   * @example
+   * const adapter = new StandardAdapter()
+   */
+  constructor () {
+    super()
+
+    /** @type {String} Adapter name */
+    this.id = this.constructor.id
```

Comment:
Interesting. I wonder if we should have a class called `BaseAdapter` that extends `EventEmitter` that does this out of the box, and the the adapters extend that one instead.

Reply:
I was thinking the same for a while, but that'll make creating third-party adapters a little messy. The ultimate lazy would be to just use the constructor.name, but that might break (i.e. if people name their constructors / classes the same), so just using costructor.id (without the instance property) would be the most minimal, while still being explicit I guess.